File: BitWiseManipulation/Java/Divide.java

import java.util.Scanner;

public class Divide {
    public static void divide(int dividend, int divisor) {
        int sign = ((dividend < 0) ^ (divisor < 0)) ? -1 : 1;
        dividend = Math.abs(dividend);
        divisor = Math.abs(divisor);

        int quotient = 0;
        while (dividend >= divisor) {
            dividend -= divisor;
            ++quotient;
        }

        System.out.println(sign * quotient);
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the dividend and divisor: ");
        int dividend = sc.nextInt();
        int divisor = sc.nextInt();

        if(divisor == 0) {
            System.out.println("Division by zero is not allowed");
            return;
        }
        divide(dividend, divisor);


    }
}

File: BitWiseManipulation/C++/Divide.cpp
#include <iostream>
#include <cstdlib>
using namespace std;

int divide(int a, int b) {
    int sign = ((a < 0) ^ (b < 0)) ? -1 : 1;
    a = abs(a);
    b = abs(b);
    int quotient = 0;
    while (a >= b) {
        a -= b;
        quotient++;
    }
    return sign * quotient;
}

int main() {
    int a, b;
    cout << "Enter dividend and divisor: ";
    cin >> a >> b;
    if (b == 0) {
        cout << "Division by zero is not allowed." << endl;
        return 1;
    }
    cout << "Quotient: " << divide(a, b) << endl;
    return 0;
}
